### ISSUE TITLE
Upstream bug: `onnxruntime==1.16.0` requires a parameter meant for GPUs (even though SCT uses the CPU version)

### ISSUE THREAD

**Issue by mguaypaq** [spinalcordtoolbox/spinalcordtoolbox#4225]
See for example this output, taken from [this testing run](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/6237705089/job/17019439874):

```
FAILED testing/api/test_deepseg_gm.py::TestCore::test_segment_volume - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/api/test_deepseg_lesion.py::test_segment - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/api/test_deepseg_sc.py::test_deep_segmentation_spinalcord[params0] - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/api/test_deepseg_sc.py::test_deep_segmentation_spinalcord[params1] - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/cli/test_cli_sct_deepseg_gm.py::test_sct_deepseg_gm_check_dice_coefficient_against_groundtruth - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/cli/test_cli_sct_deepseg_lesion.py::test_sct_deepseg_lesion_no_checks - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/cli/test_cli_sct_deepseg_sc.py::test_sct_deepseg_sc_check_output_qform_sform - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
FAILED testing/cli/test_cli_sct_deepseg_sc.py::test_sct_deepseg_sc_qc_report_exists - ValueError: This ORT build has ['AzureExecutionProvider', 'CPUExecutionProvider'] enabled. Since ORT 1.9, you are required to explicitly set the providers parameter when instantiating InferenceSession. For example, onnxruntime.InferenceSession(..., providers=['AzureExecutionProvider', 'CPUExecutionProvider'], ...)
```

It appeared in the past few days, probably due to a new version of `onnxruntime`? I'll investigate tomorrow, unless someone beats me to it.

**Comment by mguaypaq** [comment]
It looks like this tracked upstream: https://github.com/microsoft/onnxruntime/issues/17631. According to [the changelog](https://github.com/Microsoft/onnxruntime/releases/tag/v1.16.0) (see under _Known Issues_) this considered a bug, and a fix is in progress. Until it's released, I'll pin `onnxruntime` to be `<1.16`.

**Comment by mguaypaq** [comment]
Let's revisit the version pin when we prepare SCT v6.2.
